HiringApp/HiringAppCore/Providers/API/Endpoints.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Endpoints.swift
+//  Pods
+//
+//  Created by Alba Luján on 26/6/17.
+//
+//
+
+import Foundation
+import BSWFoundation
+
+enum AppEndpoints {
+    case technologies
+    case candidate(parameters: [String : AnyObject])
+    case authenticate
+    case topics(technologyId: Int)
+}
+
+extension AppEndpoints: Endpoint {
+    
+    var path: String {
+        switch self {
+        case .technologies:
+            return "/technologies"
+        case .candidate:
+            return "/contact"
+        case .authenticate:
+            return "/users"
+        case .topics(let technologyId):
+            return "/technologies/\(technologyId)/topics"
+        }
+    }
+    
+    var method: HTTPMethod {
+        switch self {
+        case .technologies, .topics:
+            return .GET
+        case .candidate, .authenticate:
+            return .POST
+        }
+    }
+    
+    var httpHeaderFields: [String : String]? {
+        switch self {
+        case .candidate:
+            return ["Content-Type": "application/json"]
+        case .technologies, .topics, .authenticate:
+            return nil
+        }
+    }
+    
+    var parameters: [String : AnyObject]? {
+        switch self {
+        case .authenticate:
+            return ["deviceId": UIDevice.current.identifierForVendor?.uuidString as AnyObject]
```

Comment:
It's out of range for an endpoint class to access SDK APIs like UIDevice. This parameter must arrive from outside, i. e., it should be an associated property in the .authenticate enum member